relentless/optimize/method.py

```diff
@@ -314,35 +323,78 @@ def __init__(self, abs_tol, max_iter, step_size, line_search=None):
         self.step_size = step_size
         self.line_search = line_search
 
+    def descent_amount(self, gradient):
+        r"""Calculate the descent amount for the optimization.
+
+        The descent amount is:
+
+        .. math::
+
+            d = \alpha\lVert\nabla f\left(\mathbf{x}\right)\rVert
+
+        Parameters
+        ----------
+        gradient : :class:`~relentless._collections.KeyedArray`
+            The gradient of the objective function.
+
+        Returns
+        -------
+        :class:`~relentless._collections.KeyedArray`
+            The descent amount, keyed on the objective function design variables.
+
+        """
+        k = _collections.KeyedArray(keys=gradient.keys)
+        try:
+            k.update(self.step_size)
+        except TypeError:
+            for i in k:
+                k[i] = self.step_size
+        return k*gradient.norm()
+
     def optimize(self, objective):
-        """Perform the steepest descent optimization for the given objective function.
+        r"""Perform the steepest descent optimization for the given objective function.
+
+        A single steepest descent update is performed as:
+
+        .. math::
+
+            \mathbf{x}_{n+1} = \mathbf{x}_{n}-d\frac{\nabla f\left(\mathbf{x}\right)}
+                                                    {\lVert\nabla f\left(\mathbf{x}\right)\rVert}
+
+        :math:`d` is the descent amount returned by :meth:`descent_amount()`.
```

Comment:
I'd prefer the `alpha*grad` version of the update rather than `d*grad/norm`. The latter looks like fixed step descent unless you knew that` d = alpha*norm`. And this is consistent with the documentation for the class.